packages/harmony-provider/src/provider.ts

```diff
@@ -0,0 +1,77 @@
+import { JsonRpcProvider, Network, Networkish, TransactionResponse } from '@ethersproject/providers'
+import { getNetwork } from './network'
+
+const HARMONY_MAINNET_URL = 'https://api.harmony.one'
+const HARMONY_TESTNET_URL = 'https://api.s0.b.hmny.io'
+
+export enum HarmonyProviderMethods {
+  getBlockNumber = 'getBlockNumber',
+  getBlock = 'getBlock',
+  getTransaction = 'getTransaction',
+}
+
+interface blockParams {
+  blockTag?: string | number
+  blockHash?: string
+}
+
+export class HarmonyProvider extends JsonRpcProvider {
+  constructor(_network?: Networkish) {
+    const network = getNetwork(_network)
+    const baseUrl = HarmonyProvider.getBaseUrl(network)
+
+    super(baseUrl, network)
+    this._nextId = 1
+  }
+
+  async perform(method: string, params: Record<string, any>): Promise<any> {
+    switch (method) {
+      case HarmonyProviderMethods.getBlock:
+        return this.getBlockByTagOrHash(params, params.includeTransactions)
+      case HarmonyProviderMethods.getBlockNumber:
+        return this.getBlockNumber()
+      case HarmonyProviderMethods.getTransaction:
+        return this.getTransaction(params.transactionHash)
+      default:
+        return super.perform(method, params)
+    }
+  }
+
+  async getTransaction(hash: string): Promise<TransactionResponse> {
+    return this.send('hmy_getTransactionByHash', [hash])
+  }
+
+  async getBlockByTagOrHash(params: blockParams, includeTransactions = false): Promise<any> {
```

Comment:
I would call this `getBlock` and try to conform to the interface from `ethers.js`:
```
getBlock(blockHashOrBlockTag: BlockTag | string | Promise<BlockTag | string>): Promise<Block>
```

    
 We can also include another method `getBlockWithTransactions` so that we avoid the boolean flag, which could be hard to parse